torch/nn/modules/module.py

```diff
@@ -1322,20 +1332,31 @@ def _maybe_warn_non_full_backward_hook(self, inputs, result, grad_fn):
                               "behavior.")
 
     def register_forward_pre_hook(
-        self, hook: Callable[..., None], prepend: bool = False
+        self,
+        hook: Callable[..., None],
+        *,
+        prepend: bool = False,
+        with_kwargs: bool = False,
     ) -> RemovableHandle:
         r"""Registers a forward pre-hook on the module.
 
         The hook will be called every time before :func:`forward` is invoked.
         It should have the following signature::
 
-            hook(module, input) -> None or modified input
+            hook(module, args) -> None or modified input
+
+        If ``with_kwargs`` is false or not specified, the input contains only
+        the positional arguments given to the module. Keyword arguments won't be
+        passed to the hooks and only to the ``forward``. The hook can modify the
+        input. User can either return a tuple or a single modified value in the
+        hook. We will wrap the value into a tuple if a single value is returned
+        (unless that value is already a tuple).
+
+        If ``with_kwargs`` is true, the forward pre hook will be passed the
+        kwargs given to the forward function and be expected to return them
+        possibly modified:
```

Comment:
Maybe change to "... given to the forward function. And if the hook modifies the input, both the args and kwargs should be returned."